compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -225,7 +225,20 @@ object SymUtils:
 
     /** The typeRef with wildcard arguments for each type parameter */
     def rawTypeRef(using Context) =
-      self.typeRef.appliedTo(self.typeParams.map(_ => TypeBounds.emptyPolyKind))
+      // Change `pre.O.this` to `pre.O` if `O` is an object
+      // This is required to avoid owner crash in ExplicitOuter.
+      // See tests/pos/i10769.scala
+      val fixPrefix = new TypeMap {
```

Comment:
I think this change makes sense, but it's weird to do this for `rawTypeRef` but not `typeRef` and `termRef` (this one is also called from `ordinalBody` when a child is an object), either we should do it for all of them (but we might run into cycles) or we need new method(s) for this concept.

Reply:
The usage of `.typeRef` and `.termRef` are very common, and not all of them need such substitution, only those that use it in trees.

>  or we need new method(s) for this concept.

This is a good idea. What about we call it `SymDenotation.typeRefForTree` or `SymUtils.typeRefForTree`?

Reply:
maybe `reachableTypeRef` to emphasize that the type can be reached through a path from the current ctx.owner?